extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -304,28 +305,92 @@ class LifecycleBucketProcessor {
                 return cb();
             }
 
-            let task;
-            // TODO: ZENKO-4501 Check if bucket is indexed before using new lifecycle improvement solution
-            if (this._lcConfig.bucketProcessor.forceLegacyListing) {
-                task = new LifecycleTask(this);
-            } else {
-                task = new LifecycleTaskV2(this);
-            }
+            return this._createLifecycleTask(backbeatMetadataProxy, params, (err, task) => {
+                if (err) {
+                    this._log.error('failed to create lifecycle bucket task', {
+                        bucket,
+                        owner,
+                        error: err,
+                    });
+                    return cb(err);
+                }
 
-            this._log.info('scheduling new task for bucket lifecycle', {
-                method: 'LifecycleBucketProcessor._processBucketEntry',
-                bucket,
-                owner,
-                details: result.details,
-                taskName: task.constructor.name,
+                this._log.info('scheduling new task for bucket lifecycle', {
+                    method: 'LifecycleBucketProcessor._processBucketEntry',
+                    bucket,
+                    owner,
+                    details: result.details,
+                    taskName: task.constructor.name,
+                });
+                return this._internalTaskScheduler.push({
+                    task,
+                    rules: config.Rules,
+                    value: result,
+                    s3target: s3,
+                    backbeatMetadataProxy,
+                }, cb);
             });
-            return this._internalTaskScheduler.push({
-                task,
-                rules: config.Rules,
-                value: result,
-                s3target: s3,
-                backbeatMetadataProxy,
-            }, cb);
+        });
+
+    }
+
+    _createLifecycleTask(backbeatMetadataProxy, params, cb) {
```

Comment:
I don't know how expensive it is to list indexes before each listing call. Is there a way to do this check only once before the first listing of the chain?

Reply:
indexing should not be expensive to query since most of it would be in memory, but there could be some reads from disk; this could also done as part on the conductor instead, so I  will check if it possible to perform the query there as opposed to checking per task in the bucket processor